### PR TITLE
[Many] Bump to min minSdkVersion of 19 across all plugins

### PR DESCRIPTION
**Pull request by gmackall**
* Bumps any `minSdkVersion`'s below 19 to 19, as the change to only support 19+ has been in stable for a bit now.
* Bumps the minimum required flutter version to 3.16 for the changed packages, as that is the first version in which we started automigrating `minSdkVersion`'s less than 19.
* Also bumps the `minSdkVersion` in the legacy project to `flutter.minSdkVersion`, as this would happen when running an app with `minSdkVersion` < 19.
* Removes some code branches for cases of sdk < 19.
* Fixes some failing tests in `video_player` - for full transparency, I have now idea how these could have been passing before?

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [ ] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Review by reidbaker — APPROVED** [review]

**Comment by stuartmorgan** [comment]
> as the change to only support 19+ has been in stable for a bit now

What's the Flutter version that has that change? We need to make sure none of the packages being changed here have a min Flutter SDK version lower than that.

**Comment by gmackall** [comment]
> What's the Flutter version that has that change? We need to make sure none of the packages being changed here have a min Flutter SDK version lower than that.

If I had to pick a specific PR, it would be https://github.com/flutter/flutter/pull/129729, though the change that really matters is the change to https://docs.flutter.dev/reference/supported-platforms, which isn't versioned along with the Flutter SDK really.

But https://github.com/flutter/flutter/pull/129729 is in 3.16 at the earliest, so it sounds like this will need to wait until the next stable, at which the minimum allowed will be N-2=3.16 (according to https://github.com/flutter/flutter/wiki/Updating-Packages-repo-for-a-stable-release)?

**Comment by stuartmorgan** [comment]
> If I had to pick a specific PR, it would be [flutter/flutter#129729](https://github.com/flutter/flutter/pull/129729), though the change that really matters is the change to https://docs.flutter.dev/reference/supported-platforms, which isn't versioned along with the Flutter SDK really.

For the purposes of the packages repo it is the PR that matters, not the policy on a wiki. The goal here is to make package changes in a way that does not suddenly break people's ability to compile their app when getting minor or bugfix version updates to the packages they use. Actively breaking people with non-breaking-version plugin changes and then pointing them to a wiki page they had no reason to look at if they file a bug would be a bad user experience. Automatically getting the last working version is a much better experience.

So when changing minimum OS versions the goal is to make is such that the `pub` resolver *will not resolve* a version that has dropped support for an OS version that the developer's current version of Flutter still supports building for. The PR that auto-migrates their project is the point at which their version of Flutter no longer supports building it.

> But [flutter/flutter#129729](https://github.com/flutter/flutter/pull/129729) is in 3.16 at the earliest, so it sounds like this will need to wait until the next stable, at which the minimum allowed will be N-2=3.16

Packages aren't required to support N-2, they just aren't allowed to support anything older; only supporting back to current `stable` is required. But packages can't *break* people on N-1 or N-2, so if you are making a change that would break compilation for people on N-2, then you need to set the minimum Flutter SDK constraint for the package(s) in question constraint to N-1 as part of the change.

https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#supported-flutter-versions

File: packages/espresso/pubspec.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ description: Java classes for testing Flutter apps using Espresso.
   Allows driving Flutter widgets from a native Espresso test.
 repository: https://github.com/flutter/packages/tree/main/packages/espresso
 issue_tracker: https://github.com/flutter/flutter/issues?q=is%3Aissue+is%3Aopen+label%3A%22p%3A+espresso%22
-version: 0.3.0+7
+version: 0.3.0+8
 
 environment:
   sdk: ^3.1.0
```
Comment (stuartmorgan) [inline packages/espresso/pubspec.yaml:9]: Based on the discussion in comments, every package being touched here that still lists Flutter 3.13 support needs to be changed to 3.16 instead.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

**Comment by stuartmorgan** [comment]
The Dart min SDK version needs to be consistent; you may want to use the `update-min-sdk` command mentioned on https://github.com/flutter/flutter/wiki/Updating-Packages-repo-for-a-stable-release to do this. You can run it with `--base-branch main --run-on-changed-packages` (assuming you branched from `main`)

**Comment by gmackall** [comment]
I ended up making the change by hand - the script made changes to every package, I believe because I had changed the legacy test.

**Comment by stuartmorgan** [comment]
Ah, yes. I get bitten by that occasionally with the version bump script; I need to make a new flag that is only for these local tool usages.

**Review by stuartmorgan — APPROVED** [review]
LGTM

**Pull request merged**